app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1175,22 +1176,41 @@ public void updateMarker(boolean isBookmarked, Place place, @Nullable fr.free.nr
         addPlaceToNearbyList(place);
 
         VectorDrawableCompat vectorDrawable;
-        if (isBookmarked) {
-            vectorDrawable = VectorDrawableCompat.create(
-                    getContext().getResources(), R.drawable.ic_custom_bookmark_marker, getContext().getTheme()
-            );
-        } else if (!place.pic.trim().isEmpty()) {
-            vectorDrawable = VectorDrawableCompat.create( // Means place has picture
-                    getContext().getResources(), R.drawable.ic_custom_map_marker_green, getContext().getTheme()
-            );
+        if (!place.pic.trim().isEmpty()) {
+            if (isBookmarked) {
```

Comment:
all of these don't need 2 execution branches on isBookmarked, a ternary will make it look all the nicer

```
vectorDrawable = VectorDrawableCompat.create(
                    getContext().getResources(), 
                    isBookmarked ? R.drawable.ic_custom_map_marker_green_bookmarked : R.drawable.ic_custom_map_marker_green,
                    getContext().getTheme()
                );
```

and we are repeating this vector creation a lot too I'd propose
```
vectorDrawable(  isBookmarked 
? R.drawable.ic_custom_map_marker_green_bookmarked 
: R.drawable.ic_custom_map_marker_green)
```

hmm actually all this logic does is choose a drawable really, maybe just a method to do that?

```
private @DrawableRes Int getIconFor(Place place, Boolean isBookmarked)
```